builtin/mainmenu/tab_online.lua

```diff
@@ -285,14 +319,46 @@ local function search_server_list(input)
 	-- Search the serverlist
 	local search_result = {}
 	for i, server in ipairs(serverlistmgr.servers) do
-		local name_matches, description_matches = true, true
-		for _, keyword in ipairs(keywords) do
+		local name_matches, description_matches, filter_matches = true, true, true
+
+		-- Check if keyword found
+		for _, keyword in ipairs(query.keywords) do
 			name_matches = name_matches and not not
 					(server.name or ""):lower():find(keyword, 1, true)
 			description_matches = description_matches and not not
 					(server.description or ""):lower():find(keyword, 1, true)
 		end
-		if name_matches or description_matches then
+
+		-- Check if mods found
+		if server.mods then
+			for _, mod in ipairs(query.mods) do
+				if table.indexof(server.mods, mod) < 0 then
```

Comment:
I think you can get rid of the outer `if server.mods then ... elseif ... end` if you just do

```suggestion
				if table.indexof(server.mods or {}, mod) < 0 then
```

here. Same below for clients.

Reply:
Changed.